Doc/library/sqlite3.rst

```diff
@@ -1224,6 +1252,45 @@ Connection objects
 
       .. versionadded:: 3.11
 
+   .. attribute:: autocommit
+
+      This attribute controls :pep:`249`-compliant transaction behaviour.
+      :attr:`!autocommit` has three allowed values:
+
+      * ``False``: Select :pep:`249`-compliant transaction behaviour,
+        implying that :mod:`!sqlite3` ensures a transaction is always open.
+        Use :meth:`commit` and :meth:`rollback` to close transactions.
+
+        This is the recommended value of :attr:`!autocommit`.
+
+      * ``True``: Use SQLite's `autocommit mode`_.
+        :meth:`commit` and :meth:`rollback` have no effect in this mode.
```

Comment:
```suggestion
        :meth:`commit` and :meth:`rollback` have no effect in this mode.

        .. note::

           The :pep:`249`-compliant :attr:`!autocommit` attribute and the
           SQLite `autocommit mode`_ are two related but distinct concepts.
           Use :attr:`in_transaction` to query the low-level SQLite
           autocommit mode.
```

Move the note here, where its most relevant and avoid an undesirable double-note. Also, tweak the text to clarify what "autocommit feature" we're talking about here, specifically the `autocommit` attribute required by PEP 249.

Reply:
How about including it in the existing note, but as a separate paragraph?

<details>
<summary>Diff</summary>

```diff
diff --git a/Doc/library/sqlite3.rst b/Doc/library/sqlite3.rst
index 3f488c37ce..a367a54903 100644
--- a/Doc/library/sqlite3.rst
+++ b/Doc/library/sqlite3.rst
@@ -1279,16 +1279,14 @@ Connection objects
 
       .. note::
 
-         The :pep:`249`-compliant autocommit feature and the
+         The :attr:`isolation_level` attribute has no effect unless
+         :attr:`autocommit` is :data:`LEGACY_TRANSACTION_CONTROL`.
+
+         The :pep:`249`-compliant :attr:`!autocommit` attribute and the
          SQLite `autocommit mode`_ are two related but distinct concepts.
          Use :attr:`in_transaction` to query the low-level SQLite
          autocommit mode.
 
-      .. note::
-
-         The :attr:`isolation_level` attribute has no effect unless
-         :attr:`autocommit` is :data:`LEGACY_TRANSACTION_CONTROL`.
-

```

</details>

Reply:
I'm personally not as fond of concatenating the two, since they are on two distinct topics, and readers may read the first few lines and assume the rest isn't relevant. But its not the end of the world...it would look better than two back to back notes, even though not as conceptually/semantically pure.

One alternative approach would be to use an `important` admonition for the "no effect" one, which would be perhaps more appropriate, and put it first, while retaining the `note` admonition for the second...assuming the Python Docs Theme actually properly gives these different colors/styles.

Reply:
Hm. Perhaps a better place for the note in question would be in the explanation (_Transaction control via the autocommit attribute_).

Reply:
Yeah, great point—that's a much better idea, as it makes a lot more sense in an explanation vs a reference.